src/CMakeLists.txt

```diff
@@ -642,7 +642,10 @@ set(ceph_common_deps
   ${CRYPTO_LIBS}
   ${CMAKE_THREAD_LIBS_INIT}
   ${CMAKE_DL_LIBS}
-  ${UDEV_LIBRARIES})
+if(LINUX)
```

Comment:
i don't think this works like C macro. probably we can have something like
```cmake
if(LINUX)
  list(APPEND ceph_common_deps ${UDEV_LIBRARIES})
endif()
```

Reply:
oops, fixed

Reply:
used HAVE_UDEV too

Reply:
Seems like HAVE_UDEV is still around and used:
```
./cmake/modules/Findudev.cmake:22:# UDEV_INCLUDE_DIR - where to find libudev.h, etc.
./cmake/modules/Findudev.cmake:23:# UDEV_LIBRARIES - List of libraries when using udev.
./cmake/modules/Findudev.cmake:24:# UDEV_FOUND - True if udev found.
./cmake/modules/Findudev.cmake:26:find_path(UDEV_INCLUDE_DIR libudev.h)
./cmake/modules/Findudev.cmake:27:find_library(UDEV_LIBRARIES udev)
./cmake/modules/Findudev.cmake:30:find_package_handle_standard_args(udev DEFAULT_MSG UDEV_LIBRARIES UDEV_INCLUDE_DIR)
./cmake/modules/Findudev.cmake:33:  UDEV_LIBRARIES
./cmake/modules/Findudev.cmake:34:  UDEV_INCLUDE_DIR)
./CMakeLists.txt:185:  set(HAVE_UDEV ${UDEV_FOUND})
./CMakeLists.txt:189:  set(HAVE_UDEV OFF)
./src/CMakeLists.txt:1063:    target_link_libraries(krbd ${KEYUTILS_LIBRARIES} ${UDEV_LIBRARIES})
./src/librbd/CMakeLists.txt:155:if(HAVE_UDEV)
```